src/test/java/org/sagebionetworks/bridge/services/MasterSchedulerServiceTest.java

```diff
@@ -0,0 +1,202 @@
+import com.google.common.collect.ImmutableList;
+
+public class MasterSchedulerServiceTest {
+    private static final long TIME_STAMP = DateTime.parse("2018-03-27T18:30-07:00").getMillis();
+    private static final String SCHEDULE_ID = "test-schedule-id";
+    private static final String CRON_SCHEDULE = "testCronSchedule";
+    private static final String UPDATED_CRON_SCHEDULE = "updatedCronSchedule";
+    private static final String SQS_QUEUE_URL = "testSysQueueUrl";
+    private static final long VERSION = 1L;
+    
+    @Spy
+    private MasterSchedulerService schedulerService;
+    
+    @Mock
+    private MasterSchedulerConfigDao configDao;
+    
+    @Mock
+    private MasterSchedulerStatusDao statusDao;
+    
+    @Captor
+    private ArgumentCaptor<MasterSchedulerConfig> configCaptor;
+    
+    @BeforeMethod
+    public void before() {
+        MockitoAnnotations.initMocks(this);
+        
+        schedulerService.setSchedulerConfigDao(configDao);
+        schedulerService.setSchedulerStatusDao(statusDao);
+    }
+    
+    @Test
+    public void testGetAllSchedulerConfigs() {
+        List<MasterSchedulerConfig> mockConfigs = ImmutableList.of(MasterSchedulerConfig.create(), MasterSchedulerConfig.create());
+        when(configDao.getAllSchedulerConfig()).thenReturn(mockConfigs);
+        
+        List<MasterSchedulerConfig> configs = schedulerService.getAllSchedulerConfigs();
+        assertEquals(configs.size(), 2);
+        
+        verify(configDao).getAllSchedulerConfig();
+    }
+    
+    @Test
+    public void testCreateSchedulerConfig() {
+        ObjectNode objNode = JsonNodeFactory.instance.objectNode();
+        objNode.put("a", true);
+        objNode.put("b", "string");
+        
+        MasterSchedulerConfig mockConfig = TestUtils.getMasterSchedulerConfig();
+        
+        when(configDao.createSchedulerConfig(configCaptor.capture())).thenReturn(mockConfig);
+        
+        schedulerService.createSchedulerConfig(mockConfig);
+        MasterSchedulerConfig config = configCaptor.getValue();
+        
+        assertEquals(config.getScheduleId(), SCHEDULE_ID);
+        assertEquals(config.getCronSchedule(), CRON_SCHEDULE);
+        assertTrue(config.getRequestTemplate().get("a").booleanValue());
+        assertEquals(config.getRequestTemplate().get("b").textValue(), "string");
+        assertEquals(config.getSqsQueueUrl(), SQS_QUEUE_URL);
+        assertEquals(config.getVersion(), new Long(VERSION));
+        assertSame(mockConfig, config);
+        
+        verify(configDao).createSchedulerConfig(configCaptor.capture());
+    }
+    
+    @Test(expectedExceptions = InvalidEntityException.class)
+    public void testCreateSchedulerConfigValidates() {
+        MasterSchedulerConfig mockConfig = TestUtils.getMasterSchedulerConfig();
+        mockConfig.setScheduleId(null);
+        
+        schedulerService.createSchedulerConfig(mockConfig);
+        verify(configDao, never()).createSchedulerConfig(any());
```

Comment:
Note: If createSchedulerConfig() throws, the verify will never get called. If you want to verify that the dao is never called, the best thing to do is to wrap it in a try catch:
```
try {
  schedulerService.createSchedulerConfig(mockConfig);
  fail("expected exception");
} catch (InvalidEntityException ex) {
  // expected exception
}
verify(configDao, never()).createSchedulerConfig(any());
```